### ISSUE TITLE
Elixir 1.4: "expected a keyword list as the second argument" in Slack.Web.Reactions.add

### ISSUE THREAD

**Issue by jdpopkin** [BlakeWilliams/Elixir-Slack#99]
My code:
```
  def handle_event(message = %{type: "message"}, slack, state) do
    target_channel = System.get_env("SLACK_CHANNEL")
    case message.channel do
      ^target_channel -> handle_message(message, slack)
      _ -> nil
    end

    {:ok, state}
  end
  def handle_event(_, _, state), do: {:ok, state}

  defp handle_message(message = %{text: _}, _slack) do
    is_track_link = Regex.match?(~r'https://(open|play).spotify.com/track/', message.text)

    if is_track_link do
      options = %{channel: message.channel, timestamp: message.ts, token: System.get_env("SLACK_TOKEN")}
      Slack.Web.Reactions.add('musical_note', options)
      nil
    end
  end
```

When this code runs on Elixir 1.4.0, I get this error:
```11:11:51 PM web.1 |  23:11:51.179 [error] expected a keyword list as the second argument, got: [{"name", 'musical_note'}]
11:11:51 PM web.1 |  23:11:51.179 [error]     (elixir) lib/keyword.ex:589: Keyword.merge/2
11:11:51 PM web.1 |      (slack) lib/slack/web/web.ex:44: Slack.Web.Reactions.add/2
11:11:51 PM web.1 |      (botify) lib/botify/slack_rtm.ex:25: Botify.SlackRtm.handle_message/2
11:11:51 PM web.1 |      (botify) lib/botify/slack_rtm.ex:12: Botify.SlackRtm.handle_event/3
11:11:51 PM web.1 |      (slack) lib/slack/bot.ex:123: Slack.Bot.websocket_handle/3
11:11:51 PM web.1 |      (websocket_client) /Users/jacksonpopkin/Developer/usr/src/me/elixir/botify/deps/websocket_client/src/websocket_client.erl:448: :websocket_client.handle_websocket_frame/2
11:11:51 PM web.1 |      (stdlib) gen_fsm.erl:451: :gen_fsm.handle_msg/7
11:11:51 PM web.1 |      (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
11:11:51 PM web.1 |  23:11:51.208 [error] ** Websocket client Slack.Bot terminating in :websocket_handle/3
11:11:51 PM web.1 |     for the reason :error:%RuntimeError{message: "expected a keyword list as the second argument, got: [{\"name\", 'musical_note'}]"}
```

But if I run `brew switch elixir 1.3.4` and run again, this works as I would expect it to - my Slack bot reacts with :musical_note:.

(I know I'm passing a map and not a keyword, but passing a keyword here doesn't seem to work on either version of Elixir.)

I'm very new to Elixir; is it possible that this change from the [1.4 release notes](https://github.com/elixir-lang/elixir/releases/tag/v1.4.0) has broken Slack.Web.Reactions.add? `[Macro] Do not print aliases as keys inside keyword lists in Macro.to_string/2`

**Comment by mjs2600** [comment]
@squaresurf and I are looking into this. We should have a PR in a few hours.

**Comment by BlakeWilliams** [comment]
@mjs2600 Does #98 fix this for you?

**Comment by squaresurf** [comment]
Oh man, I didn't see #98 so I opened #100. The main difference is that #100 adds test coverage and solves the issue in a slightly different manner. I also left out the warning fixes in order to make sure the fix for the error was clear. I'm planning on opening another PR that fixes all the elixir 1.4.0 warnings.

**Comment by ashkan18** [comment]
Thanks guys, I just added similar test to my https://github.com/BlakeWilliams/Elixir-Slack/pull/98 as well.

@squaresurf it seems your test is failing because of using `describe` which was something that was added in Elixir 1.3 I believe and in travis we also run 1.2

**Comment by squaresurf** [comment]
Thank you @ashkan18. I fixed up #100 to pass tests as well. I also opened #101 for elixir compile time warnings.

**Comment by squaresurf** [comment]
@BlakeWilliams do you have any idea when you'll release a new version with the fix for this? We're currently blocked on upgrading to elixir v1.4.0 until this gets updated.

**Comment by BlakeWilliams** [comment]
Sorry, merging a PR then cutting a release right after. :)

**Comment by BlakeWilliams** [comment]
0.9.3 should be live now

**Comment by squaresurf** [comment]
Thank you! 🎉
